terraform/setup/iam.tf

```diff
@@ -0,0 +1,81 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+resource "aws_iam_role" "cwagent_role" {
+  name = module.common.cwa_iam_role
+
+  assume_role_policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+          "Sid": "",
+          "Effect": "Allow",
+          "Principal": {
+            "Service": "ec2.amazonaws.com"
+          },
+          "Action": "sts:AssumeRole"
+            
+        },
+        {
+          "Sid": "",
+          "Effect": "Allow",
+          "Principal": {
+            "Service": "ecs-tasks.amazonaws.com"
+          },
+          "Action": "sts:AssumeRole"
+        }
+    ]
+}
+EOF
+}
+
+data "aws_iam_policy_document" "user-managed-policy-document" {
+  statement {
+    actions = [
+      "cloudwatch:GetMetricData",
+      "cloudwatch:PutMetricData",
+      "cloudwatch:ListMetrics",
+      "cloudwatch:GetMetricStatistics",
+      "ec2:DescribeVolumes",
+      "ec2:DescribeTags",
+      "ec2:DescribeInstances",
+      "logs:*",
+      "dynamodb:DescribeTable",
+      "dynamodb:PutItem",
+      "dynamodb:CreateTable",
+      "dynamodb:Query",
+      "dynamodb:UpdateItem",
+      "ecs:DescribeTasks",
+      "ecs:ListTasks",
+      "ecs:DescribeContainerInstances",
+      "ecs:DescribeServices",
+      "ecs:ListServices",
+      "ecs:DescribeTaskDefinition",
+      "ecr:GetAuthorizationToken",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage",
+      "ssm:Describe*",
+      "ssm:Get*",
+      "ssm:List*",
+      "s3:*",
```

Comment:
For S3 I am not sure. However, questions for everyone, would it be better to use * for all resources in AWS or would it be better to list the specific actions in the resources? I would prefer the former one since it would make the dev less awareness regarding the actions being used but want your thought on that before implementing it.